### DISCUSSION TITLE
Compute geometric measures

### DISCUSSION THREAD

**Discussion by samgam1981** [cnr-isti-vclab/meshlab#1296] [Q&A]
HI could you please let me know where to find the source code for the filter-computer geometric measures please,
Need to understand the principles used to calculate the principal axes.

Thanks

**Comment by jmespadero** [comment]
Hi samgam,

The method that is executed to compute the geometric measures is:

https://github.com/cnr-isti-vclab/meshlab/blob/3a1d8f2afd805a7f2d73ca08558b71c5ba5388c7/src/meshlabplugins/filter_measure/filter_measure.cpp#L373

Inside that method, there is a call to computePrincipalAxisCloud(const CMeshO& m) which do the work:

https://github.com/cnr-isti-vclab/meshlab/blob/3a1d8f2afd805a7f2d73ca08558b71c5ba5388c7/src/meshlabplugins/filter_measure/filter_measure.cpp#L690

Basically
1. Compute in `bp` the barycenter/centroid of the vertices of the mesh (Average position)
2. Compute in `cov` the 3x3 matrix of covariances of the coordinates. The actual code of computing the matrix is at https://github.com/cnr-isti-vclab/vcglib/blob/a443e903a66e14a44b170056eee90d233cd753f9/vcg/math/matrix33.h#LL392
3. Return the Eigen Vectors of matrix `cov`
